src/models/auditModelGenerator.js

```diff
@@ -29,26 +29,38 @@ const pgSetConfigIfNull = (settingName, value, alias) => `
     true
   ) as "${alias}"`;
 
+const auditedTransactions = new Set();
+
 const addAuditTransactionSettings = async (sequelize, instance, options, type) => {
   const loggedUser = httpContext.get('loggedUser') ? httpContext.get('loggedUser') : '';
   const transactionId = httpContext.get('transactionId') ? httpContext.get('transactionId') : '';
   const sessionSig = httpContext.get('sessionSig') ? httpContext.get('sessionSig') : '';
   const auditDescriptor = httpContext.get('auditDescriptor') ? httpContext.get('auditDescriptor') : '';
 
-  const statements = [
-    pgSetConfigIfNull('audit.loggedUser', loggedUser, 'loggedUser'),
-    pgSetConfigIfNull('audit.transactionId', transactionId, 'transactionId'),
-    pgSetConfigIfNull('audit.sessionSig', sessionSig, 'sessionSig'),
-    pgSetConfigIfNull('audit.auditDescriptor', auditDescriptor, 'auditDescriptor'),
-  ];
+  if (!auditedTransactions.has(transactionId)) {
+    auditedTransactions.add(transactionId);
+    const statements = [
+      pgSetConfigIfNull('audit.loggedUser', loggedUser, 'loggedUser'),
+      pgSetConfigIfNull('audit.transactionId', transactionId, 'transactionId'),
+      pgSetConfigIfNull('audit.sessionSig', sessionSig, 'sessionSig'),
+      pgSetConfigIfNull('audit.auditDescriptor', auditDescriptor, 'auditDescriptor'),
+    ];
 
-  if (loggedUser !== '' || transactionId !== '' || auditDescriptor !== '') {
-    await sequelize.queryInterface.sequelize.query(
-      `SELECT
-        '${type}' "Type",
-        ${statements.join(',')}
-      `.replace(/[\r\n]+/gm, ' '),
-    );
+    if (loggedUser !== '' || transactionId !== '' || auditDescriptor !== '') {
+      await sequelize.queryInterface.sequelize.query(
+        `SELECT
+          '${type}' "Type",
+          ${statements.join(',')}
+        `.replace(/[\r\n]+/gm, ' '),
+      );
+    }
+  }
+};
+
+const removeFromAuditedTransactions = async () => {
+  const transactionId = httpContext.get('transactionId') ? httpContext.get('transactionId') : '';
+  if (auditedTransactions.has(transactionId)) {
+    auditedTransactions.delete(transactionId);
```

Comment:
minor: you don't need to wrap this call to `delete` in a `has` conditional, set won't freak out if you pass a key that doesn't exist to `delete`